src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -63,36 +63,53 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     addTermRating(request);
     response.setContentType("text/html; charset=UTF-8");
     response.setCharacterEncoding("UTF-8");
-    response.sendRedirect("/index.html");
   }
 
   public void addTermRating(HttpServletRequest request) throws IOException {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    String termFeedback = request.getParameter("term-input");
-    Long termRating = Long.parseLong(request.getParameter("rating-term"));
-    Long workHours = Long.parseLong(request.getParameter("hoursOfWork"));
-    Long difficulty = Long.parseLong(request.getParameter("difficulty"));
-    String professorFeedback = request.getParameter("prof-input");
-    Long professorRating = Long.parseLong(request.getParameter("rating-professor"));
+
+    StringBuilder jb = new StringBuilder();
+    String line = null;
+    try {
+      BufferedReader reader = request.getReader();
+      while ((line = reader.readLine()) != null) jb.append(line);
+    } catch (Exception exception) {
+      /*report an error*/
+    }
+
+    try {
+      JSONObject jsonObject = new JSONObject(jb.toString());
+      schoolName = jsonObject.getString("school-name");
+      courseName = jsonObject.getString("course-name");
+      profName = jsonObject.getString("prof-name");
+      termName = jsonObject.getString("term");
+      units = (long) jsonObject.getFloat("units");
+      termFeedback = jsonObject.getString("term-input");
+      professorFeedback = jsonObject.getString("prof-input");
+      termRating = (long) jsonObject.getFloat("rating-term");
+      professorRating = (long) jsonObject.getFloat("rating-prof");
+      workHours = (long) jsonObject.getFloat("hours");
+      difficulty = (long) jsonObject.getFloat("difficulty");
+    } catch (JSONException exception) {
+      // If it could not parse string.
+      throw new IOException("Error parsing JSON request string");
+    }
 
     float termScore = getSentimentScore(termFeedback);
     float professorScore = getSentimentScore(professorFeedback);
 
-    // Gets user ID from URL.
-    String userId = request.getParameter("ID");
-    // Gets term key from URL.
-    Key currentTermKey = KeyFactory.stringToKey(request.getParameter("term"));
+    // Quick change, will modify tests as well after the demo.
```

Comment:
It's post demo now. Is this still the change we want to submit?

Reply:
Yes! This is still the change we want to submit, I am re-formatting my tests cases on a separate PR that I am working on now.